src/avram/query_builder.cr

```diff
@@ -21,6 +21,19 @@ class Avram::QueryBuilder
     [statement] + args
   end
 
+  # Prepares the SQL statement by combining the `args` and `statement`
+  # in to a single `String`
+  def to_prepared_sql : String
+    params = args.map { |arg| "'#{String.new(PQ::Param.encode(arg).slice)}'" }
```

Comment:
This needs the single quotes because strings replaced in the SQL would be `name = Don` instead of `name = 'Don'`. Thankfully ALL postgres values can contain the single quotes and still work fine. 

Now, if we can figure out how to properly cast the value, then we can do `value.class.adapter.to_db(value)`. Right now that throws a compile error because the compiler thinks `value` is a Union type 😕